packages/lesswrong/server/ckEditor/ckEditorCallbacks.ts

```diff
@@ -96,16 +96,10 @@ defineQuery({
     //  * The logged-in user is the post author
     //  * The logged in user is an admin or moderator (or otherwise has edit permissions)
 
-    // User must exist
-    if (!currentUser?._id) {
-      throw new Error("No current user with _id");
-    }
-
     if (
       (post.shareWithUsers && _.contains(post.shareWithUsers, currentUser?._id))
-      || (linkSharingEnabled(post)
-          && (!canonicalLinkSharingKey || keysMatch))
-      || (linkSharingEnabled(post) && post.linkSharingKeyUsedBy?.includes(currentUser?._id))
+      || (linkSharingEnabled(post) && (!canonicalLinkSharingKey || keysMatch))
+      || (linkSharingEnabled(post) && (currentUser && post.linkSharingKeyUsedBy?.includes(currentUser._id)))
```

Comment:
This condition is only salient for logged-in users; we now need to check that `currentUser` exists since `.includes` can't run on something that might be `undefined` (and we're no longer checking that `currentUser` exists at the very top of the query).